File: tools/jjbmc/src/main/java/jjbmc/ErrorLogger.java
/* This file is part of jmltoolkit project - https://github.com/jmltoolkit
 * jmltk is licensed under the Lesser GNU General Public License Version 2 and Apache License
 * SPDX-License-Identifier: LGPL-3.0-or-later Apache-2.0
 */
package jjbmc;

public class ErrorLogger {
    private static final long START_TIME = System.currentTimeMillis();
    public static final String RESET = "\033[0m"; // Text Reset
    public static final String RED_BOLD = "\033[1;31m"; // RED
    public static final String GREEN_BOLD = "\033[1;32m"; // GREEN
    public static final String YELLOW_BOLD = "\033[1;33m"; // YELLOW

    private static boolean debugOn = false;

    private static void print(String fmt, Level level, Object... args) {
        if (level == Level.DEBUG && !debugOn) return;

        System.out.printf("%05d \033[1;%dm[%s]\033[0m ", System.currentTimeMillis() - START_TIME, 32, level);
        System.out.printf(fmt, args);
        System.out.println();
    }

    public static void debug(String fmt, Object... args) {
        print(fmt, Level.DEBUG, args);
    }

    public static void debug(Throwable obj) {
        print("%s", Level.DEBUG, obj);
    }

    public static void info(Object fmt, Object... args) {
        print(fmt.toString(), Level.INFO, args);
    }

    public static void fatal(String fmt, Object... args) {
        print(fmt, Level.FATAL, args);
    }

    public static void error(String fmt, Object... args) {
        print(fmt, Level.ERROR, args);
    }

    public static void warn(String fmt, Object... args) {
        print(fmt, Level.WARN, args);
    }

    public static void setDebugOn() {
        debugOn = true;
    }
}

File: tools/jjbmc/src/main/kotlin/jjbmc/ErrorLogger.kt
/* This file is part of jmltoolkit project - https://github.com/jmltoolkit
 * jmltk is licensed under the Lesser GNU General Public License Version 2 and Apache License
 * SPDX-License-Identifier: LGPL-3.0-or-later Apache-2.0
 */
package jjbmc

object ErrorLogger {
    private val START_TIME = System.currentTimeMillis()
    const val RESET: String = "\u001b[0m" // Text Reset
    const val RED_BOLD: String = "\u001b[1;31m" // RED
    const val GREEN_BOLD: String = "\u001b[1;32m" // GREEN
    const val YELLOW_BOLD: String = "\u001b[1;33m" // YELLOW

    private var debugOn = false

    private fun print(fmt: String, level: Level?, vararg args: Any?) {
        if (level == Level.DEBUG && !debugOn) return

        System.out.printf("%05d \u001b[1;%dm[%s]\u001b[0m ", System.currentTimeMillis() - START_TIME, 32, level)
        System.out.printf(fmt, *args)
        println()
    }

    fun debug(fmt: String, vararg args: Any?) {
        print(fmt, Level.DEBUG, *args)
    }

    fun debug(obj: Throwable?) {
        print("%s", Level.DEBUG, obj)
    }

    fun info(fmt: Any, vararg args: Any?) {
        print(fmt.toString(), Level.INFO, *args)
    }

    fun fatal(fmt: String, vararg args: Any?) {
        print(fmt, Level.FATAL, *args)
    }

    fun error(fmt: String, vararg args: Any?) {
        print(fmt, Level.ERROR, *args)
    }

    fun warn(fmt: String, vararg args: Any?) {
        print(fmt, Level.WARN, *args)
    }

    fun setDebugOn() {
        debugOn = true
    }
}
